master/internal/api_workspace.go

```diff
@@ -159,23 +202,39 @@ func (a *apiServer) GetWorkspaces(
 	if err != nil {
 		return nil, err
 	}
+
+	curUser, _, err := grpcutil.GetUser(ctx, a.m.db, &a.m.config.InternalConfig.ExternalSessions)
```

Comment:
since `GetWorkspaces` already gets the current user for the get_workspaces query earlier (for pinned workspaces), you could move this `grpcutil.GetUser` up earlier , or keep and use your new `UserFromProto` function